### PR TITLE
Object Oriented JavaScript

### PR DESCRIPTION
**Pull request by Eugene-CG**
### Object oriented javascript

[Demo](https://eugene-cg.github.io/frontend-nanodegree-arcade-game/)
[Code base](https://github.com/Eugene-CG/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    ) {
+      setTimeout(() => {
+        player.x = tileSize.width * 2;
+        player.y = tileSize.height * 4 - charactersSize.charboy;
+      }, 100);
+    }
+  }
+  update(dt) {
+    this.x += this.speed * dt;
+    if (this.x > 500) this.x = -tileSize.width;
```
Comment (A-Ostrovnyy) [inline submissions/Eugene-CG/OOJS-Frogger-Game/app.js:50]: What is the magic number 500?

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+const enemyLocationY = [
+  tileSize.height - charactersSize.bug,
+  tileSize.height * 2 - charactersSize.bug,
+  tileSize.height * 3 - charactersSize.bug,
+];
+const allEnemies = [];
+
+enemyLocationY.forEach((location) =>
```
Comment (A-Ostrovnyy) [inline submissions/Eugene-CG/OOJS-Frogger-Game/app.js:94]: `.forEach` + `.push` you can replace with `.map` or `.reduce`

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Awesome!

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js (hunk 1 above)
Comment (Eugene-CG) [reply submissions/Eugene-CG/OOJS-Frogger-Game/app.js:50]: > What is the magic number 500?

Can I change code in github or better do it from vscode?
I wondering how to deal with this minor changes better

Comment (A-Ostrovnyy) [reply submissions/Eugene-CG/OOJS-Frogger-Game/app.js:50]: It's up to you)

**Comment by Eugene-CG** [comment]
Little mistake in last commit comment -_-
Don`t know how to change this using github, only git --amend -m "..."  =|

**Review by Eugene-CG — COMMENTED** [review]
Great Jobe Myself

**Comment by Eugene-CG** [comment]
> Great Jobe Myself Забавна штука, я теж так можу, хе :3 а як видалити =Ґ

**Comment by A-Ostrovnyy** [comment]
@Eugene-CG why you close this PR?

**Comment by Eugene-CG** [comment]
> @Eugene-CG why you close this PR?

Sorry, I want to delete my comment and click everything I see

**Comment by A-Ostrovnyy** [comment]
To avoid this situation I always use my code editor instead of web interface) don't worry about the same commit message, if you want to change it use `--amend`

**Comment by Eugene-CG** [comment]
> To avoid this situation I always use my code editor instead of web interface) don't worry about the same commit message, if you want to change it use `--amend`

Already changed :3

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+  update() {}
+}
+class Enemy extends Character {
+  constructor(x, y, speed) {
+    super(x, y, "images/enemy-bug.png");
+    this.speed = speed;
+  }
+  collision() {
```
Comment (OleksiyRudenko) [inline submissions/Eugene-CG/OOJS-Frogger-Game/app.js:32]: By convention, method (and function) names should start with a verb.

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    super(x, y, "images/char-boy.png");
+  }
+  reset() {
+    if (this.y < tileSize.height - charactersSize.charboy) {
+      setTimeout(() => {
+        alert("You won");
+        this.y = tileSize.height * 4 - charactersSize.charboy;
+        this.x = tileSize.width * 2;
+      }, 100);
+    }
```
Comment (OleksiyRudenko) [inline submissions/Eugene-CG/OOJS-Frogger-Game/app.js:65]: The method name is imperative, however the body assumes a condition. This confuses me as a peer developer.
You may want either rename the method or do conditional checks elsewhere where coordinates change happens and let this method just set some coordinates imperatively.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Eugene-CG great job!
A couple of improvements and we are done.
You may also want to check yourself against the checklist the Kottacheck both will add below in the conversation.

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/Eugene-CG/OOJS-Frogger-Game/app.js]: Still not fixed :cry:

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    this.x += this.speed * dt;
+    if (this.x > tileSize.width * 5) this.x = -tileSize.width;
+    this.checkCollision();
+  }
+}
+class Player extends Character {
+  constructor(x, y) {
+    super(x, y, "images/char-boy.png");
+  }
+  reset() {
```
Comment (A-Ostrovnyy) [inline submissions/Eugene-CG/OOJS-Frogger-Game/app.js:62]: Fix method name)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
One last issue)

File: submissions/Eugene-CG/OOJS-Frogger-Game/app.js (hunk 5 above)
Comment (Eugene-CG) [reply submissions/Eugene-CG/OOJS-Frogger-Game/app.js]: > 

Changed from "reset" to "resetToStartPosition"

Comment (Eugene-CG) [reply submissions/Eugene-CG/OOJS-Frogger-Game/app.js]: :3

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**